crates/wasmtime/src/trampoline.rs

```diff
@@ -77,7 +77,7 @@ fn create_handle(
                 externref_activations_table: store.externref_activations_table()
                     as *const VMExternRefActivationsTable
                     as *mut _,
-                stack_map_registry: store.stack_map_registry() as *const StackMapRegistry as *mut _,
+                stack_map_lookup: Some(std::mem::transmute(store.stack_map_lookup())),
```

Comment:
This seems to compile for me:

```rust
trait A: 'static {}

fn foo(a: Option<*const dyn A>) {}

fn bar(a: &dyn A) {
    foo(Some(a));
}
```

so is this `transmute` required? (same for the one in `instance.rs`)

Reply:
Adding the `'static` constraint to the trait did the trick, so I'll remove the transmute